src/core/utils/heap.hpp

```diff
@@ -37,6 +37,8 @@
 
 #include "openthread-core-config.h"
 
+#if !OPENTHREAD_CONFIG_EXTERNAL_HEAP_ENABLE
```

Comment:
also need to add `#if !OPENTHREAD_CONFIG_EXTERNAL_HEAP_ENABLE ... #endif` to `heap.cpp`?